File: ejemplos_en_clase/sockets/soquete.rb
#!/usr/bin/ruby
# coding: utf-8
require 'socket'

server = TCPServer.new 2000

def procesa(client)
  while true
    data = client.gets(16)
    if data
      print "Me dijeron: %s" % data
      client.puts("OK!\n")
    else
      print "Fin de comunicación"
      return 0
    end
  end
end

loop do
  client = server.accept
  procesa(client)
  client.close
end

File: ejemplos_en_clase/sockets/soquete.py
#!/usr/bin/python
# *-* Encoding: utf-8

import socket
import sys
import os

server = '/tmp/socket'
try:
  os.unlink(server)
except OSError:
  pass

sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
sock.bind(server)

sock.listen(1)

while True:
  connection, client = sock.accept()
  while True:
    data = connection.recv(10)
    if data:
      print "Me dijeron: %s" % data
      connection.send("OK!\n")
    else:
      print "Fin de comunicación"
      break
